src/test/java/com/example/music_player/storage/CloudStorageAmazonS3Test.java

```diff
@@ -67,21 +78,43 @@ public void saveWhenBucketNotExist() {
 
         verify(s3Client, times(1)).createBucket(bucketName);
         verify(s3Client, times(1)).putObject(any(PutObjectRequest.class));
-        assertThat(fileObject.delete()).isEqualTo(false);
     }
 
     @Test
-    public void saveWhenBucketExistNow() {  //TODO Nullpointer s3Client in CloudStorageAmazonS3, How to properly mock it?
+    public void saveWhenBucketExistNow() {
         when(s3Client.doesBucketExistV2(bucketName)).thenReturn(true);
         when(s3Client.putObject(any())).thenReturn(putObjectResult);
 
         cloudStorageAmazonS3.save(inputStream, "originalFilename", "contentType");
 
         verify(s3Client, never()).createBucket(bucketName);
         verify(s3Client, times(1)).putObject(any(PutObjectRequest.class));
```

Comment:
Btw temporary file should be deleted in positive scenarios too, so verification of testedFile.delete() is useful here as well as it is in negative scenarios.

Reply:
in this key, will be use Power Mokito